### PR TITLE
[expo-cli] add --skip-credentials-check option

### PR DESCRIPTION
**Pull request by satya164**
## Why

We want people to be able to skip checking for remote credentials.

### CODE REVIEW

File: packages/expo-cli/src/commands/eas-build/build/credentials.ts [hunk 1]
```diff
@@ -20,7 +21,10 @@ async function ensureCredentialsAutoAsync(
       }
     case Workflow.Generic: {
       const hasLocal = await provider.hasLocalAsync();
-      const hasRemote = await provider.hasRemoteAsync();
+      const hasRemote = skipCredentialsCheck ? false : await provider.hasRemoteAsync();
```
Comment (wkozyra95) [inline packages/expo-cli/src/commands/eas-build/build/credentials.ts:24]: `skipCredentialsCheck` should skip validation of credentials, but we still need have those values present.

`ensureCredentials` should not contain any logic related to that flag, for now, only place where we have credentials validation is Ios remote credentials

File: packages/expo-cli/src/credentials/provider/iOSCredentialsProvider.ts [hunk 2]
```diff
@@ -77,14 +82,29 @@ export default class iOSCredentialsProvider implements CredentialsProvider {
     return await credentialsJsonReader.readIosCredentialsAsync(this.projectDir);
   }
   private async getRemoteAsync(): Promise<iOSCredentials> {
-    await runCredentialsManager(this.ctx, new SetupIosBuildCredentials(this.app));
+    if (!this.app.skipCredentialsCheck) {
+      log('Skipping credentials check');
```
Comment (dsokal) [inline packages/expo-cli/src/credentials/provider/iOSCredentialsProvider.ts:86]: Something seems wrong with this `if`. Either the message is inaccurate or the condition should be `if (this.app.skipCredentialsCheck))`.

File: packages/expo-cli/src/credentials/provider/AndroidCredentialsProvider.ts [hunk 3]
```diff
@@ -29,7 +30,7 @@ export default class AndroidCredentialsProvider implements CredentialsProvider {
 
   public async initAsync() {
     await this.ctx.init(this.projectDir, {
-      nonInteractive: this.ctx.nonInteractive,
+      nonInteractive: this.options.nonInteractive,
```
Comment (wkozyra95) [inline packages/expo-cli/src/credentials/provider/AndroidCredentialsProvider.ts:33]: :+1:

File: packages/expo-cli/src/credentials/provider/iOSCredentialsProvider.ts [hunk 4]
```diff
@@ -77,14 +82,29 @@ export default class iOSCredentialsProvider implements CredentialsProvider {
...
+      if (this.app.skipCredentialsCheck) {
+        throw new Error(
+          'Distribution certificate is missing and credentials check was skipped. Run without --skip-credentials-check to set it up.'
+        );
+      } else {
+        throw new Error('Distribution certificate is missing');
+      }
     }
     const provisioningProfile = await this.ctx.ios.getProvisioningProfile(this.app);
     if (!provisioningProfile) {
```
Comment (wkozyra95) [inline packages/expo-cli/src/credentials/provider/iOSCredentialsProvider.ts:100]:
```suggestion
    if (!provisioningProfile?.provisioningProfile) {
```

File: packages/expo-cli/src/credentials/provider/iOSCredentialsProvider.ts [hunk 5]
```diff
@@ -77,14 +82,29 @@ export default class iOSCredentialsProvider implements CredentialsProvider {
     return await credentialsJsonReader.readIosCredentialsAsync(this.projectDir);
   }
   private async getRemoteAsync(): Promise<iOSCredentials> {
-    await runCredentialsManager(this.ctx, new SetupIosBuildCredentials(this.app));
+    if (!this.app.skipCredentialsCheck) {
+      log('Skipping credentials check');
+      await runCredentialsManager(this.ctx, new SetupIosBuildCredentials(this.app));
+    }
     const distCert = await this.ctx.ios.getDistCert(this.app);
     if (!distCert) {
```
Comment (wkozyra95) [inline packages/expo-cli/src/credentials/provider/iOSCredentialsProvider.ts:90]:
```suggestion
    if (!distCert?.certP12 || !distCert?.certPassword) {
```

**Review by wkozyra95 — APPROVED** [review]

File: packages/expo-cli/src/credentials/provider/iOSCredentialsProvider.ts [hunk 6]
```diff
@@ -15,16 +15,21 @@ export interface iOSCredentials {
...
+  skipCredentialsCheck: boolean;
+}
+
 export default class iOSCredentialsProvider implements CredentialsProvider {
   public readonly platform = 'ios';
   private readonly ctx = new Context();
   private credentials?: iOSCredentials;
 
-  constructor(private projectDir: string, private app: AppLookupParams) {}
+  constructor(private projectDir: string, private app: Options) {}
```
Comment (wkozyra95) [inline packages/expo-cli/src/credentials/provider/iOSCredentialsProvider.ts:28]: I would prefer not to merge those values, pass 3 arguments to constructor instead

File: packages/expo-cli/src/credentials/provider/AndroidCredentialsProvider.ts [hunk 7]
```diff
@@ -11,25 +11,29 @@ export interface AndroidCredentials {
   keystore: Keystore;
 }
 
-interface Options {
+interface Params {
```
Comment (wkozyra95) [inline packages/expo-cli/src/credentials/provider/AndroidCredentialsProvider.ts:14]:
```suggestion
interface AppLookupParams {
```

File: packages/expo-cli/src/credentials/provider/AndroidCredentialsProvider.ts [hunk 8]
```diff
@@ -11,25 +11,29 @@ export interface AndroidCredentials {
...
+interface Options {
+  nonInteractive: boolean;
+}
+
 export default class AndroidCredentialsProvider implements CredentialsProvider {
   public readonly platform = 'android';
   private readonly ctx = new Context();
 
-  constructor(private projectDir: string, private options: Options) {}
+  constructor(private projectDir: string, private params: Params, private options: Options) {}
```
Comment (wkozyra95) [inline packages/expo-cli/src/credentials/provider/AndroidCredentialsProvider.ts:27]:
```suggestion
  constructor(private projectDir: string, private app: AppLookupParams, private options: Options) {}
```

File: packages/expo-cli/src/credentials/provider/AndroidCredentialsProvider.ts [hunk 9]
```diff
@@ -11,25 +11,29 @@ export interface AndroidCredentials {
...
 
   private get projectFullName(): string {
-    const { projectName, accountName } = this.options;
+    const { projectName, accountName } = this.params;
```
Comment (wkozyra95) [inline packages/expo-cli/src/credentials/provider/AndroidCredentialsProvider.ts:30]:
```suggestion
    const { projectName, accountName } = this.app;
```

**Review by wkozyra95 — APPROVED** [review]

**Review by wkozyra95 — APPROVED** [review]

**Review by dsokal — APPROVED** [review]

**Comment by dsokal** [comment]
Looks good, please update the changelog! :)

**Pull request merged**